pkg/doc/verifiable/credential.go

```diff
@@ -465,18 +473,25 @@ func NewCredential(vcData []byte, opts ...CredentialOpt) (*Credential, error) {
 	return cred, nil
 }
 
-func decodeRaw(vcData []byte, crOpts *credentialOpts) (vcDataDecoded []byte, raw *rawCredential, err error) {
+func decodeRaw(vcData []byte, crOpts *credentialOpts) ([]byte, *rawCredential, error) {
 	if crOpts.issuerPublicKeyFetcher != nil {
```

Comment:
Agree. I also have doubts about clarity of configurations.
Now it works like this:
- if `issuerPublicKeyFetcher` is defined, try to deserialize JWS, if error, abort
- if `unsecuredJWT` is true, try to deserialize unsecured JWT, if error, abort
- otherwise, unmarshal from JSON.

I created a follow-up issue #445  for this, just because current PR is already too big.